src/language-js/print/ternary.js

```diff
@@ -153,6 +153,91 @@ function printTernaryTest(path, options, print) {
   return printed;
 }
 
+const rightSideNodeNamesSet = new Set(["right", "init", "argument"]);
```

Comment:
If I didn't misunderstand this, we are checking NodeA is NodeB's `.right` or `.init`, but not checking NodeB's type. I don't like this part, this may lead to unexpect cases.

For example `argument`, I guess you mean to check argument of `ReturnStatement`/`YieldExpression` etc, let's list them.